concent_api/core/tasks.py

```diff
@@ -29,12 +29,25 @@
 logger = logging.getLogger(__name__)
 
 
-@shared_task
+@shared_task(bind=True)
 @log_task_errors
 @transaction.atomic(using='control')
-def upload_finished(subtask_id: str):
+def upload_finished(self, subtask_id: str):
+    # Worker locks database row corresponding to the subtask in the subtask table.
     try:
-        subtask = Subtask.objects.get(subtask_id=subtask_id)
+        subtask = Subtask.objects.select_for_update(nowait=True).get(subtask_id=subtask_id)
+    except DatabaseError:
+        logging.warning(
+            f'Subtask object with ID {subtask_id} database row is locked, '
+            f'retrying task {self.request.retries}/{self.max_retries}'
```

Comment:
The message is misleading. It can fail also for other reasons.

We still want to retry then but the error message should be different. You should definitely log the original message with `logger.exception()`.

Reply:
done